RelationBehavior.php

```diff
@@ -204,7 +206,8 @@ protected function loadData()
                         }
                         // create new junction models
                         foreach ($data['data'] as $relatedModelId) {
-                            $junctionModel = new $junctionModelClass();
+                            $junctionModel = new $junctionModelClass(array_merge(!ArrayHelper::isAssociative($via->on) ? [] : $via->on,
```

Comment:
Why you pass empty array instead throw RelationException with message 'ON condition for attribute ' . $attribute . ' must be associative array' like in 167 line?

You should add checking on condition for via in 167 line, instead put empty array.